src/tools/pkcs11-tool.c

```diff
@@ -5363,6 +5455,18 @@ static int test_decrypt(CK_SESSION_HANDLE sess)
 		printf("No OpenSSL support, unable to validate decryption\n");
 #else
 		for (n = 0; n < num_mechs; n++) {
+			switch (mechs[n]) {
+			case CKM_RSA_PKCS:
+			case CKM_RSA_PKCS_OAEP:
+			case CKM_RSA_X_509:
```

Comment:
`CKM_RSA_X_509` can be used for encryption/decryption, but you do not handle it above. Does it work?

Reply:
Unfortunately, it is not working.
I will try to fix this issue during the week.

Reply:
does it work now?

Reply:
ping?